src/gossip/fuzz_service.zig

```diff
@@ -63,11 +63,22 @@ pub fn randomPongPacket(random: std.rand.Random, keypair: *const KeyPair, to_add
     return packet;
 }
 
-pub fn randomSignedGossipData(random: std.rand.Random, must_pass_sign_verification: bool) !SignedGossipData {
-    const keypair = try KeyPair.create(null);
-    const pubkey = Pubkey.fromPublicKey(&keypair.public_key);
-    const lci = LegacyContactInfo.default(if (must_pass_sign_verification) pubkey else Pubkey.initRandom(random));
-    return SignedGossipData.initSigned(&keypair, .{ .LegacyContactInfo = lci });
+pub fn randomSignedGossipData(random: std.rand.Random, maybe_should_pass_sig_verification: ?bool) !SignedGossipData {
+    var keypair = try KeyPair.create(null);
+    const pubkey = try Pubkey.fromPublicKey(&keypair.public_key);
+
+    // will have random id
+    var value = try SignedGossipData.randomWithIndex(random, &keypair, 0);
+    value.data.LegacyContactInfo = LegacyContactInfo.default(try Pubkey.fromPublicKey(&keypair.public_key));
+    try value.sign(&keypair);
+
+    const should_pass_sig_verification = maybe_should_pass_sig_verification orelse random.boolean();
+    if (should_pass_sig_verification) {
+        value.data.setId(pubkey);
+        try value.sign(&keypair);
+    }
+
+    return value;
```

Comment:
Wrt to the argument against the `sign` method, this, for example, seems like a downgrade compared to the original code: it doesn't even generated random data, it overwrites the randomly generated data with the default value, and redundantly re-sets the id.